Classes/Controller/AdministrationController.php

```diff
@@ -25,15 +31,222 @@ public function __construct(
+        'formPageId' => $formPageId,
+        'ip' => $ip,
+        'formName' => $formName,
+        'startDate' => $startDate ? new \DateTime($startDate) : null,
+        'endDate' => $endDate ? new \DateTime($endDate) : null,
+        'itemsPerPage' => $itemsPerPage,
+      ],
     ]);
 
     return $moduleTemplate->renderResponse('Administration/Index');
   }
+
+  public function selectFieldsAction(?string $logDataUids = null, string $filetype = ''): ResponseInterface {
+    $moduleTemplate = $this->moduleTemplateFactory->create($this->request);
+
+    $logEntries = $this->logRepository->findByUids(array_map('intval', explode(',', $logDataUids ?? '')));
+    $fieldsToShow = [
+      'Global fields' => [
+        'ip',
+        'form_name',
+        'submission_date',
+        'form_page_id',
+      ],
+      'Custom fields' => [],
+      'System fields' => [
+        'key_hash',
+        'unique_hash',
+      ],
+    ];
+
+    foreach ($logEntries as $logDataRow) {
+      $serializedArray = unserialize($logDataRow->getParams());
+      // fallback. Use empty array in case the serialized data is invalid
+      // TODO: add better error handling. Maybe exit & display error?
+      if (!is_array($serializedArray)) {
+        $serializedArray = [];
+      }
+
+      $params = $this->prepareFlatArray($serializedArray);
+      $fields = array_keys($params);
+      foreach ($fields as $idx => $rowField) {
+        if (!in_array($rowField, $fieldsToShow['Custom fields'])) {
+          $fieldsToShow['Custom fields'][] = $rowField;
+        }
+      }
+    }
+
+    $moduleTemplate->assignMultiple([
+      'fieldsToShow' => $fieldsToShow,
+      'fileType' => $filetype,
+      'logDataUids' => $logDataUids,
+    ]);
+
+    return $moduleTemplate->renderResponse('Administration/SelectFields');
+  }
+
+  /**
+   * @param array<int, array<string, string>> $array
+   * @param array<string>                     $headers
+   */
+  protected function arrayToCsvConvert(array $array, array $headers, string $filename = 'export.csv', string $delimiter = ','): void {
+    if (count($array) < 1 || count($array[0]) !== count($headers)) {
+      return;
+    }
+
+    $csvFile = fopen('php://memory', 'w');
+    if ($csvFile) {
+      fputcsv($csvFile, $headers, $delimiter);
+      foreach ($array as $line) {
+        // remove linebreaks
+        $line = array_map(function ($field) {
+          return str_replace(["\r", "\n"], '', $field);
+        }, $line);
+
+        fputcsv($csvFile, $line, $delimiter);
+      }
+
+      fseek($csvFile, 0);
+      header('Content-Type: text/csv');
+      header('Content-Disposition: attachment; filename="'.$filename.'";');
+      fpassthru($csvFile);
+    }
+
+    $response = $this->responseFactory->createResponse(200);
+
+    throw new PropagateResponseException($response);
+  }
+
+  /**
+   * TODO: add ability to map language files to forms. Fieldkeys like 1.customer.email should get changed to their translation.
```

Comment:
for this to work the log needs to also log the form config ts

Reply:
I dont know if thats even neccessary. Form configs dont have to specifiy their language files anymore, so that proably wouldnt even be enough